src/packet_buffer.cpp

```diff
@@ -53,15 +53,15 @@ namespace libtorrent {
 	}
 #endif
 
-	void* packet_buffer_impl::insert(index_type idx, void* value)
+	packet_ptr packet_buffer::insert(index_type idx, packet_ptr value)
 	{
 		INVARIANT_CHECK;
 
 		TORRENT_ASSERT_VAL(idx <= 0xffff, idx);
 		// you're not allowed to insert NULLs!
 		TORRENT_ASSERT(value);
 
-		if (value == nullptr) return remove(idx);
+		if (!value) return remove(idx);
```

Comment:
I don't think you can compare a ``unique_ptr`` against a raw pointer or ``nullptr`` directly. But it has a bool conversion operator